c10/core/impl/PyInterpreter.h

```diff
@@ -30,6 +30,46 @@ using Stack = std::vector<c10::IValue>;
 namespace c10 {
 namespace impl {
 
+struct C10_API PyInterpreter;
+
+struct C10_API CUDATraceFunctionWrapper {
+  using event_creation_sig = void(const PyInterpreter*, uintptr_t event);
+  using event_deletion_sig = void(const PyInterpreter*, uintptr_t event);
+  using event_record_sig =
+      void(const PyInterpreter*, uintptr_t event, uintptr_t stream);
+  using event_wait_sig =
+      void(const PyInterpreter*, uintptr_t event, uintptr_t stream);
+  using memory_allocation_sig = void(const PyInterpreter*, uintptr_t pointer);
+  using memory_deallocation_sig = void(const PyInterpreter*, uintptr_t pointer);
+  using stream_creation_sig = void(const PyInterpreter*, uintptr_t stream);
+
+  event_creation_sig* event_creation_fn_;
+  event_deletion_sig* event_deletion_fn_;
+  event_record_sig* event_record_fn_;
+  event_wait_sig* event_wait_fn_;
+  memory_allocation_sig* memory_allocation_fn_;
+  memory_deallocation_sig* memory_deallocation_fn_;
+  stream_creation_sig* stream_creation_fn_;
+
+  CUDATraceFunctionWrapper(
+    event_creation_sig* event_creation_fn,
+    event_deletion_sig* event_deletion_fn,
+    event_record_sig* event_record_fn,
+    event_wait_sig* event_wait_fn,
+    memory_allocation_sig* memory_allocation_fn,
+    memory_deallocation_sig* memory_deallocation_fn,
+    stream_creation_sig* stream_creation_fn)
+    : event_creation_fn_(event_creation_fn),
+      event_deletion_fn_(event_deletion_fn),
+      event_record_fn_(event_record_fn),
+      event_wait_fn_(event_wait_fn),
+      memory_allocation_fn_(memory_allocation_fn),
+      memory_deallocation_fn_(memory_deallocation_fn),
+      stream_creation_fn_(stream_creation_fn) {}
```

Comment:
Do we need this constructor? Doesn't the default one do the same thing?

Reply:
As discussed, I will leave it to be consistent with the `PyInterpreter` structure.